src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -224,22 +241,24 @@ internal static string QueryContextAuthenticationPackage(SafeDeleteContext secur
             SecurityBuffer outSecurityBuffer,
             ref ContextFlagsPal contextFlags)
         {
-            // TODO (Issue #3718): The second buffer can contain a channel binding which is not supported
-            if ((null != inSecurityBufferArray) && (inSecurityBufferArray.Length > 1))
-            {
-                throw new PlatformNotSupportedException(SR.net_nego_channel_binding_not_supported);
-            }
-
             SafeFreeNegoCredentials negoCredentialsHandle = (SafeFreeNegoCredentials)credentialsHandle;
 
             if (negoCredentialsHandle.IsDefault && string.IsNullOrEmpty(spn))
             {
                 throw new PlatformNotSupportedException(SR.net_nego_not_supported_empty_target_with_defaultcreds);
             }
 
+            SecurityBuffer cbtBuffer = null;
+            if ((inSecurityBufferArray != null) && (inSecurityBufferArray.Length > 1))
+            {
+                Debug.Assert(inSecurityBufferArray[1].type == SecurityBufferType.SECBUFFER_CHANNEL_BINDINGS);
```

Comment:
Are there cases when type is valid but not  SECBUFFER_CHANNEL_BINDINGS. If so PNSP may be option.
It does not seems so from quick read.

Reply:
If there is more than 1 buffer, the current code assumption is already that it must be a SECBUFFER_CHANNEL_BINDINGS.  That is why I have an Assert there.